src/ethereum/cancun/fork.py

```diff
@@ -527,6 +527,9 @@ def apply_body(
 
     process_message_call(system_tx_message, system_tx_env)
 
+    if account_exists_and_is_empty(state, BEACON_ROOTS_ADDRESS):
+        destroy_account(state, BEACON_ROOTS_ADDRESS)
+
```

Comment:
Doing this here means we now have two distinct places where we destroy touched accounts. I feel that makes it harder to understand. Does `BEACON_ROOTS_ADDRESS` appear in `process_message_call`'s `touched_accounts`? We could factor out account removal into its own function (shared by system transactions here and by `process_transaction`.)

Reply:
Updated